rest-openehr/src/main/java/org/ehrbase/rest/openehr/OpenehrCompositionController.java

```diff
@@ -379,30 +384,25 @@ private <T extends CompositionResponseData> Optional<InternalResponse<T>> buildC
             CompositionRepresentation responseRepresentation,
             URI uri,
             List<String> headerList,
-            Supplier<T> factory) {
+            Supplier<T> factory,
+            Optional<String> availableTemplateId) {
```

Comment:
I'm not a fan of optional as parameter because you can not enforce that the optional is not null

Reply:
I would make @Nullable String templateIdFromContext and then do a null check later